src/Badges.sol

```diff
@@ -82,6 +83,12 @@ contract Badges is
     transferOwnership(_nextOwner);
   }
 
+  function refreshMetadata(string[] memory _specUris) external onlyOwner {
+    for (uint256 i = 0; i < _specUris.length; i++) {
+      emit RefreshMetadata(_specUris[i]);
```

Comment:
I think we should check for whether the specURI exists, no? I don't think you covered this use case i.e., using invalid or unregistered specUris